tests/cpp/test_multidevice_pipeline.cpp

```diff
@@ -304,17 +304,31 @@ INSTANTIATE_TEST_SUITE_P(
         ::testing::Values(true),
         ::testing::Bool()));
 
+// Different scheduling modes used in
+// PipelineTestStagedReduction.StagedReduction
 enum class SchedulingMode {
+  // Manual interdevice scheduling, no intra-device scheduling
```

Comment:
I am not sure to understand. Why all alias op? Here the fusion features a reduction for example.
This mode simply means that we don't apply scheduling primitive (`split`, `merge`, `parallelize`, `setMemoryType`, `inlineAt`, etc) besides the fusion definition, so everything will be "by default", and basically all axis will be `ParallelType::Serial`.
What do you mean by most fusion "won't work"? It always work in all of our tests

Reply:
> basically all axis will be ParallelType::Serial

Get it. So these unscheduled intra-device fusions do "work", just super slow because they are executed by one thread in loops. Is my understanding right?

Reply:
correct!

Reply:
So back to my initial question -- what's the practical benefit of this mode? AFAICS, ReductionScheduler mode also generates functional but slow fusions, but it's strictly better because the fusions at least do reduction fast?

Reply:
by practical benefit do you mean "performance"?
The practical benefit here is just that we need to support all these situations.

> ReductionScheduler mode also generates functional but slow fusions

why slow fusion?

> but it's strictly better because the fusions at least do reduction fast?

yes

Reply:
> by practical benefit do you mean "performance"?

That or fixing functional bugs or improving developer velocity. 

> Or more precisely, the "InterDeviceOnly" mode is the only thing we support before this patch, and it is the sinmplest situation, but the slowest

That matches my understanding. I think the practical benefit of this (testing) mode is developer velocity. If we see a test failing in this mode, we are certain that some inter-device stuff (e.g. communication) is broken. 

> why slow fusion?

Non-reduction intra-device segments will run as slow as not scheduled -- serialized to one thread.

Reply:
right!